src/libraries/System.Text.Json/src/System/Text/Json/Serialization/WriteStack.cs

```diff
@@ -172,6 +192,14 @@ public void Pop(bool success)
             else
             {
                 Debug.Assert(_continuationCount == 0);
+
+                if (Current.AsyncEnumerator is not null)
+                {
+                    // we have completed serialization of an AsyncEnumerator,
+                    // pop from the stack and schedule for async disposal.
+                    PendingAsyncDisposables ??= new();
```

Comment:
nit: it's not clear what the target type is without scrolling up

```suggestion
                    PendingAsyncDisposables ??= new List<AsyncDisposable>();
```